api/wasm/v0/cpp/proxy_wasm_common.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Common enumerations available to WASM modules and shared with sandbox.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <string>
+
+enum class WasmResult : uint32_t {
+  Ok = 0,
+  // The result could not be found, e.g. a provided key did not appear in a table.
+  NotFound = 1,
+  // An argument was bad, e.g. did not not conform to the required range.
+  BadArgument = 2,
+  // A protobuf could not be serialized.
+  SerializationFailure = 3,
+  // A protobuf could not be parsed.
+  ParseFailure = 4,
+  // A provided expression (e.g. "foo.bar") was illegal or unrecognized.
+  BadExpression = 5,
+  // A provided memory range was not legal.
+  InvalidMemoryAccess = 6,
+  // Data was requested from an empty container.
+  Empty = 7,
+  // The provided CAS did not match that of the stored data.
+  CasMismatch = 8,
```

Comment:
What is a CAS in this context? Compare and swap? Spell this out?

Reply:
Done